src/mountscript.sh

```diff
@@ -596,59 +639,429 @@ ensure_aznfswatchdog()
     fi
 }
 
+#
+# Ensure azfilenfs-watchdog service is running, if not bail out with an appropriate
+# error.
+#
+ensure_azfilenfs-watchdog()
+{
+    if ! systemctl is-active --quiet azfilenfs-watchdog; then
+        eecho "azfilenfs-watchdog service not running!"
+        pecho "Start the azfilenfs-watchdog service using 'systemctl start azfilenfs-watchdog' and try again."
+        pecho "If the problem persists, contact Microsoft support."
+        return 1
+    fi
+}
+
+find_next_available_port_and_start_stunnel()
+{
+        while true
+        do
+                # get the next available port
+                available_port=$(get_next_available_port)
+                if [ $? -ne 0 ]; then
+                        eecho "Failed to get the next available port for nfsv4 mount."
+                        return 1
+                fi
+                vecho "Next Available Port: '$available_port'"
+
+                if [ -z "$available_port" ]; then
+                        eecho "Running out of ports. Nfsv4 has port range $NFSV4_PORT_RANGE_START to $NFSV4_PORT_RANGE_END. All ports from this range are used by other processes."
+                        return 1
+                fi
+
+                used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+                vecho "used port: '$used_port'"
+
+                chattr -f -i $stunnel_conf_file
+
+                sed -i "s/$used_port/$available_port/" $stunnel_conf_file
+                if [ $? -ne 0 ]; then
+                        eecho "Failed to replace the port in $stunnel_conf_file."
+                        chattr -f +i $stunnel_conf_file
+                        return 1
+                fi
+                chattr -f +i $stunnel_conf_file
+
+                new_used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+
+                # start the stunnel process
+		stunnel_status=$(stunnel $stunnel_conf_file 2>&1)
```

Comment:
why not just create the file in a location and just use systemctl start stunnel4?

Reply:
If the customer has stunnel.conf files in /etc/stunnel for some other purpose, and if we run systemctl start stunnel4, that would start all stunnel configurations. So I don't think running systemctl start stunnel4 would be a good solution in such scenario.